### PR TITLE
Floyd warshal js

### PR DESCRIPTION
**Pull request by prince-09**
Fixes https://github.com/jainaman224/Algo_Ds_Notes/issues/2205

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Floyd warshall in javascript

#### Languages Used:
- java script

#### Files Added:
- Floyd_Warshall.js

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+         });
+      });
+      return dist;
+   }
+}
+
+let g = new Graph();
+var edges = prompt("Please enter number of edges", "5");
+var vertices = prompt("Please enter number of vertices", "5");
+for(let i=0;i<vertices;i++){
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Please give space around operators.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            });
+         });
+      });
+      return dist;
+   }
+}
+
+let g = new Graph();
+var edges = prompt("Please enter number of edges", "5");
+var vertices = prompt("Please enter number of vertices", "5");
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Add a blank line after this for readability.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+for(let i=0;i<vertices;i++){
+  var e=prompt("Please enter vertex name", "A");
+  g.addNode(e);
+}
+/*g.addNode("A");
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Blank line before and after this comment block.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+g.addNode("B");
+g.addNode("C");
+g.addNode("D");*/
+for(let i=0;i<edges;i++){
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Space around =<;

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  var e1=prompt("Please enter start edge name", "A");
+  var e2=prompt("Please enter start edge name", "A");
+  var w=prompt("Please enter start edge name", 100);
+  g.addEdge(e1, e2, w);
+}
+/*g.addEdge("A", "C", 100);
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Blank line before this.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+g.addEdge("A", "B", 3);
+g.addEdge("A", "D", 4);
+g.addEdge("D", "C", 3);*/
+/* output : 
+A: {C: 7, B: 3, D: 4, A: 0}
+B: {A: 3, B: 0, C: 10, D: 7}
+C: {A: 7, D: 3, B: 10, C: 0}
+D: {A: 4, C: 3, B: 7, D: 0}*/
+console.log(g.floydWarshallAlgorithm());
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: This console.log should be before output

**Comment by prince-09** [comment]
@singh-shreya6  All changes made...please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
+   /*The Floyd Warshall Algorithm is for solving the All Pairs Shortest Path problem.
+   The problem is to find shortest distances between every pair of vertices in a given
+   edge weighted directed Graph.This is Floyd Warshall algorithm in java script.*/
+
+   class Graph {
+      constructor() {
+         this.edges = {};
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: 4 spce indentation inside constructor

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
+   /*The Floyd Warshall Algorithm is for solving the All Pairs Shortest Path problem.
+   The problem is to find shortest distances between every pair of vertices in a given
+   edge weighted directed Graph.This is Floyd Warshall algorithm in java script.*/
+
+   class Graph {
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Please remove the extra space at the start of line. Each block inside function or class or conditional statement should be indented 4 spaces. However this 3 space indentation at start of line here is unnecessary.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+      }
+
+      addNode(node) {
+         this.nodes.push(node);
+         this.edges[node] = [];
+      }
+
+      addEdge(node1, node2, weight = 1) {
+         this.edges[node1].push({ node: node2, weight: weight });
+         this.edges[node2].push({ node: node1, weight: weight });
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: 4 space

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+      }
+   floydWarshallAlgorithm() {
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Indentation + Leave a blank line before this statement.

**Comment by prince-09** [comment]
@singh-shreya6 mam indentation and PR template fixed.

**Comment by prince-09** [comment]
@singh-shreya6 ...review please

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+            // For existing edges assign the dist to be same as weight
+            this.edges[this.nodes[i]].forEach(e => (dist[this.nodes[i]][e.node] = e.weight));
+
+            this.nodes.forEach(n => {
+            // For all other nodes assign it to infinity
+            if (dist[this.nodes[i]][n] == undefined)
+                dist[this.nodes[i]][n] = Infinity;
+            // For self edge assign dist to be 0
+            if (this.nodes[i] === n) dist[this.nodes[i]][n] = 0;
+              });
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Fix indentation of this });

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        this.nodes.forEach(i => {
+                this.nodes.forEach(j => {
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: more than 4 spaces indentation

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+                        this.nodes.forEach(k => {
+        // Check if going from i to k then from k to j is better
+        // than directly going from i to j. If yes then update
+        // i to j value to the new value
+        if (dist[i][k] + dist[k][j] < dist[i][j])
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: Indent 4 spaces from the last opening bracket you see i.e k loop here.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    let g = new Graph();
+
+    var edges = prompt("Please enter number of edges", "5");
+
+    var vertices = prompt("Please enter number of vertices", "5");
+
+    for(let i = 0;i < vertices; i++){
+        var e=prompt("Please enter vertex name", "A");
+        g.addNode(e);
+        }
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: indentation of } not correct.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+
+    /*g.addNode("A");
+    g.addNode("B");
+    g.addNode("C");
+    g.addNode("D");*/
+
+        for(let i = 0;i < edges; i++){
+        var e1=prompt("Please enter start edge name", "A");
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: 4 space indentation for statements inside this block.

File: Floyd_Warshall_Algorithm/Floyd_Warshall.js [hunk 16]
```diff
@@ -0,0 +1,87 @@
+/*The Floyd Warshall Algorithm is for solving the All Pairs Shortest Path problem.
+The problem is to find shortest distances between every pair of vertices in a given
+edge weighted directed Graph.This is Floyd Warshall algorithm in java script.*/
+class Graph {
```
Comment (singh-shreya6) [inline Floyd_Warshall_Algorithm/Floyd_Warshall.js]: One blank line before this.

**Comment by prince-09** [comment]
@singh-shreya6, mam if these are the only required changes , I have made them and I have checked from my side too. Please merge it this time.

**Pull request merged**